File: PRACTICAS/EjerciciosRuby/parcela.rb
class Parcela

  @@factoralquilercalle = 1.0
  @@factoralquilercasa = 1.0
  @@factoralquilerhotel = 4.0

  def initialize (nombre,precioCompra,precioEdificar, precioBaseAlquiler)
    @nombre = nombre
    @precioCompra = precioCompra
    @precioEdificar = precioEdificar
    @precioBaseAlquiler = precioBaseAlquiler
    @numCasas = 0
    @numHoteles = 0
  end

  attr_reader :nombre, :precioCompra, :precioEdificar, :numCasas, :numHoteles

  def getPrecioAlquilerCompleto
    @precioBaseAlquiler*(@@factoralquilercalle * (@@factoralquilercasa+@numCasas+(@@factoralquilerhotel*@numHoteles)))
  end

  def construirCasa
    @numCasas+=1
    @true
  end

  def construirHoteles
    @numHoteles+=1
    @true
  end
end

p = Parcela.new("monte negro",1000,2000,400)

x = 0

p.construirCasa
p.construirHoteles


@precioalquiler = p.getPrecioAlquilerCompleto

puts "El precio del alquiler completo de la parcela #{p.nombre} es #{@precioalquiler}"

File: PRACTICAS/EjerciciosJava/src/main/java/ejercicios/Parcela.java
package ejercicios;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */


/**
 *
 * @author Blanca
 */
public class Parcela {
    private String nombre;
    private float precioCompra, precioEdificar, precioBaseAlquiler;
    private int numCasas, numHoteles;
    
    public Parcela(String nombre, float precioCompra, float precioEdificar, float precioBaseAlquiler){
        this.nombre = nombre;
        this.precioCompra = precioCompra;
        this.precioEdificar = precioEdificar;
        this.precioBaseAlquiler = precioBaseAlquiler;
        
        this.numCasas = 0;
        this.numHoteles = 0;
    }
    
    public String getNombre(){
        return nombre;
    }
    
    public float getPrecioCompra(){
        return precioCompra;
    }
    
    public int getNumCasas(){
        return numCasas;
    }
    
    public int getNumHoteles(){
        return numHoteles;
    }
    
    public float getPrecioAlquilerCompleto(){
        float precio_alquiler = precioBaseAlquiler * (1 + numCasas + (4*numHoteles));
        return precio_alquiler;
    }
    
    public boolean construirCasa(){
        numCasas++; 
        return true;
    }
    
    public boolean construirHoteles(){
        numHoteles++; 
        return true;
    }
    
    public boolean igualdadIdentidad(Parcela otraParcela){
        if (otraParcela == this) {
            return true;
        }
        return false;
    }
    
    public boolean igualdadEstado(Parcela otraParcela){
        if((this.nombre.equals(otraParcela.nombre))){
                return true;
        }
               
        return false;
    }
    
}
